Expand the developer documentation (#144)
brainglobe/brainglobe.github.io: docs/source/community/developers/new_releases.md

The `v` prefix **is necessary** as the workflow will only attempt to upload to `PyPI` if the tag matches the format previously provided.
The `X`, `Y`, and `Z` values should be integers corresponding to the new version number.

## Coordinating releases with the documentation and the metapackage

> Releases will be made ad-hoc as bug-fixes and new features become available. When releasing a new version of a BrainGlobe repository, we also need to update the website and the metapackage accordingly.

Reviewer: ```suggestion
Releases will be made ad-hoc as bug-fixes and new features become available.
When releasing a new version of a BrainGlobe repository, we also need to update the website, the metapackage, and any other tools that depend on that repository accordingly.
```

> This means we will typically have up to three dependent PRs, one each in

Reviewer: ```suggestion
This means we will typically at least three dependent PRs;
```

> - the repository itself (containing the bugfix or new feature we'd like to release)
> - the [website repository](https://github.com/brainglobe/brainglobe.github.io) (if necessary)
> - the [metapackage repository](https://github.com/brainglobe/brainglobe-meta)
> We should cross-link the latter to, and release all of them simultaneously once they are all merged into `main`.

Reviewer: ```suggestion
- One in the repository itself (containing the bugfix or new feature we'd like to release)
- One in the [website repository](https://github.com/brainglobe/brainglobe.github.io) (if necessary)
- One for each repository that depends on the updated tool, to bump the dependency version.
- One in the [metapackage repository](https://github.com/brainglobe/brainglobe-meta), which pins the new versions of all affected tools at once.

We should cross-link the latter to the website update, and release all affected packages to PyPI (and conda if appropriate) once they are all merged into `main`.
Ideally, updates and releases should be made in an order that [follows the dependency tree](./repositories/brainglobe-meta/index#dependency-tree) - starting with our lower level tools, than their dependents, then dependents of those dependents, and so on.
The meta-package itself will always be the last by this convention.
```

The `v` prefix **is necessary** as the workflow will only attempt to upload to `PyPI` if the tag matches the format previously provided.
The `X`, `Y`, and `Z` values should be integers corresponding to the new version number.

## Coordinating releases with the documentation and the metapackage

Releases will be made ad-hoc as bug-fixes and new features become available.
When releasing a new version of a BrainGlobe repository, we also need to update the website, the metapackage, and any other tools that depend on that repository accordingly.

> This means we will typically at least three dependent PRs;

Reviewer 2: ```suggestion
This means we will typically create at least three dependent PRs;
```

- One in the repository itself (containing the bugfix or new feature we'd like to release)

> - One in the [website repository](https://github.com/brainglobe/brainglobe.github.io) (if necessary)

Reviewer 2: ```suggestion
- One in the [website repository](https://github.com/brainglobe/brainglobe.github.io) (to update documentation if necessary)
```

> - One for each repository that depends on the updated tool, to bump the dependency version.

Reviewer 2: For consistency
```suggestion
- One for each repository that depends on the updated tool, to bump the dependency version
```

> - One in the [metapackage repository](https://github.com/brainglobe/brainglobe-meta), which pins the new versions of all affected tools at once.

Reviewer 2: For consistency
```suggestion
- One in the [metapackage repository](https://github.com/brainglobe/brainglobe-meta), which pins the new versions of all affected tools at once
```

We should cross-link the latter to the website update, and release all affected packages to PyPI (and conda if appropriate) once they are all merged into `main`.

> Ideally, updates and releases should be made in an order that [follows the dependency tree](./repositories/brainglobe-meta/index#dependency-tree) - starting with our lower level tools, than their dependents, then dependents of those dependents, and so on.

Reviewer 2: This link is giving me a warning. I don't think sphinx likes section references.

Reviewer: I think it will be happy if you put

`./repositories/brainglobe-meta/index.md#dependency-tree` 

for the link - the `.md` gets auto-converted to `.html` so there aren't any problems there. I've used this syntax in a number of places in this repo already without any warnings showing up.

Author: I worked around it differently, for now - but good to know :)
